sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java

```diff
@@ -1853,13 +1853,36 @@ public void canCreateVMSSWithEphemeralOSDisk() throws Exception {
             .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_18_04_LTS)
             .withRootUsername("jvuser")
             .withSsh(sshPublicKey())
-            .withUpgradeMode(UpgradeMode.AUTOMATIC)
+            .withUpgradeMode(UpgradeMode.MANUAL)
             .withEphemeralOSDisk()
             .withPlacement(DiffDiskPlacement.CACHE_DISK)
-            .withCapacity(2)
+            .withCapacity(1)
             .create();
         Assertions.assertTrue(uniformVMSS.isEphemeralOSDisk());
 
+        // somehow newly created vmss with ephemeral OS disk has outdated VMs and there's a delay before vmss detects that
+        ResourceManagerUtils.sleep(Duration.ofMinutes(1));
```

Comment:
created an issue for service: 
https://github.com/Azure/azure-rest-api-specs/issues/22552